### PR TITLE
Refactor MethodTable::ContainsStackPtr

### PR DESCRIPTION
**Pull request by jkotas**
- Rename ContainsStackPtr to IsByRefLike. It is the term used for this kind of types in ECMA spec.
- Change the check to be based on flag instead of hard coded list of types
- Remove redundant unused method of the same name on EEClass

### CODE REVIEW

**Comment by jkotas** [comment]
@gkhanna79 PTLA

**Comment by jkotas** [comment]
cc @KrzysztofCwalina This is prep-work for the `Span<T>` prototype

File: src/mscorlib/src/System.Private.CoreLib.txt [hunk 1]
```diff
@@ -1356,8 +1356,8 @@ NotImplemented_ResourcesLongerThan2^63 = Resource files longer than 2^63 bytes a
 NotSupported_NYI = This feature is not currently implemented.
 NotSupported_AbstractNonCLS = This non-CLS method is not implemented.
 NotSupported_ChangeType = ChangeType operation is not supported.
-NotSupported_ContainsStackPtr = Cannot create boxed TypedReference, ArgIterator, or RuntimeArgumentHandle Objects.
-NotSupported_ContainsStackPtr[] = Cannot create arrays of TypedReference, ArgIterator, ByRef, or RuntimeArgumentHandle Objects.
+NotSupported_ContainsStackPtr = Cannot create boxed ByRef-like values.
```
Comment (gkhanna79) [inline src/mscorlib/src/System.Private.CoreLib.txt]: Do you want to update the resource name as well, while you are at it?

Comment (jkotas) [inline src/mscorlib/src/System.Private.CoreLib.txt]: Done.

File: src/vm/clsload.cpp [hunk 2]
```diff
@@ -3697,14 +3706,14 @@ TypeHandle ClassLoader::CreateTypeHandleForTypeKey(TypeKey* pKey, AllocMemTracke
             typeHnd = TypeHandle(new(mem)  ArrayTypeDesc(templateMT, paramType));
         }
         else 
-        {            
+        {
             // no parameterized type allowed on a reference
-            if (paramType.GetInternalCorElementType() == ELEMENT_TYPE_BYREF || 
+            if (paramType.GetInternalCorElementType() == ELEMENT_TYPE_BYREF ||
```
Comment (gkhanna79) [inline src/vm/clsload.cpp]: Is there a specific reason to not do the IsByRefLike check here, similar to the above?

**Comment by gkhanna79** [comment]
LGTM - do we want to add a test for Span<T>?

**Comment by jkotas** [comment]
> do we want to add a test for Span?

Yes, eventually once it start coming online.

File: src/vm/clsload.cpp (hunk 2 above)
Comment (jkotas) [inline src/vm/clsload.cpp]: I believe that this check is too restrictive for no good reason. No need to make it even more restrictive. I will add comment about it.

**Pull request merged**